Code/Tools/AssetProcessor/native/unittests/AssetRequestHandlerUnitTests.cpp

```diff
@@ -73,11 +74,21 @@ namespace AssetProcessor
     };
 }
 
+AssetRequestHandlerUnitTests::AssetRequestHandlerUnitTests()
+    : UnitTest::AssetProcessorUnitTestBase()
+{
+}
+
+AssetRequestHandlerUnitTests::~AssetRequestHandlerUnitTests()
+{
+}
```

Comment:
Why were these needed? Especially the destructor

Reply:
The destructor is required because of the forward declaration and unique pointer for MockAssetRequestHandler and MockConnectionHandler. Otherwise the compiler generates a default one, and it needs a complete declaration of these two classes, which caused the Linux build failure: https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-12653/2/pipeline